enterprise/backend/src/metabase_enterprise/semantic_search/indexer.clj

```diff
@@ -128,9 +265,94 @@
            ^Instant last-seen-change
            ^Duration exit-early-cold-duration]}
    ^Instant now]
-  (and start-time
-       last-seen-change
-       (.isAfter now (.plus last-seen-change exit-early-cold-duration))))
+  (or (and last-seen-change
+           (.isAfter now (.plus last-seen-change exit-early-cold-duration)))
+      (and (nil? last-seen-change)
+           start-time
+           (.isAfter now (.plus start-time exit-early-cold-duration)))))
```

Comment:
changed this so that we also yield back to quartz if we haven't seen any changes since the start.

https://github.com/metabase/metabase/pull/62070/commits/a77aeb2a0fec9c7134e045717be0d206ff53cb43